op-challenger/types/game_type.go

```diff
@@ -0,0 +1,86 @@
+package types
+
+import (
+	"fmt"
+
+	"github.com/ethereum-optimism/optimism/op-service/enum"
+)
+
+// GameType is the type of dispute game
+type GameType uint8
+
+// DefaultGameType returns the default dispute game type.
+func DefaultGameType() GameType {
+	return AttestationDisputeGameType
+}
+
+// String returns the string value of a dispute game type.
+func (g GameType) String() string {
+	return DisputeGameTypes[g]
+}
+
+const (
+	// AttestationDisputeGameType is the uint8 enum value for the attestation dispute game
+	AttestationDisputeGameType GameType = iota
+	// FaultDisputeGameType is the uint8 enum value for the fault dispute game
+	FaultDisputeGameType
+	// ValidityDisputeGameType is the uint8 enum value for the validity dispute game
+	ValidityDisputeGameType
+)
+
+// DisputeGameTypes is a list of dispute game types.
+var DisputeGameTypes = []string{"attestation", "fault", "validity"}
+
+// Valid returns true if the game type is within the valid range.
+func (g GameType) Valid() bool {
+	return g >= AttestationDisputeGameType && g <= ValidityDisputeGameType
+}
+
+// DisputeGameType is a custom flag type for dispute game type.
+type DisputeGameType struct {
+	Enum     []enum.Stringered
+	Default  string
+	selected string
+}
+
+// NewDisputeGameType returns a new dispute game type.
+func NewDisputeGameType() *DisputeGameType {
+	return &DisputeGameType{
+		Enum:     enum.StringeredList(DisputeGameTypes),
+		Default:  DefaultGameType().String(),
+		selected: DefaultGameType().String(),
+	}
+}
+
+// Set sets the dispute game type.
+func (d *DisputeGameType) Set(value string) error {
+	for _, enum := range d.Enum {
+		if enum.String() == value {
+			d.selected = value
+			return nil
+		}
+	}
+
+	return fmt.Errorf("allowed values are %s", enum.EnumString(d.Enum))
+}
+
+// String returns the selected dispute game type.
+func (d DisputeGameType) String() string {
+	if d.selected == "" {
+		return d.Default
+	}
+	return d.selected
+}
+
+// Type maps the [DisputeGameType] string value to a [GameType] enum value.
+func (d DisputeGameType) Type() GameType {
+	if d.selected == DisputeGameTypes[0] {
+		return AttestationDisputeGameType
+	} else if d.selected == DisputeGameTypes[1] {
+		return FaultDisputeGameType
+	} else if d.selected == DisputeGameTypes[2] {
+		return ValidityDisputeGameType
+	} else {
+		return DefaultGameType()
+	}
```

Comment:
Couldn't we just make `d.selected` be a `GameType` and return  it directly rather than storing it as a string?  `Set` is already validating the provided string so we'd wind up with:

```
// Set sets the dispute game type.
func (d *DisputeGameType) Set(value string) error {
	for i, enum := range d.Enum {
		if enum.String() == value {
			d.selected = GameType(i)
			return nil
		}
	}

	return fmt.Errorf("allowed values are %s", enum.EnumString(d.Enum))
}

// String returns the selected dispute game type.
func (d DisputeGameType) String() string {
	return d.selected.String()
}

// Type maps the [DisputeGameType] string value to a [GameType] enum value.
func (d DisputeGameType) Type() GameType {
	return d.selected
}
```

If that doesn't work for some reason, we should add a test because they all pass once you remove a couple of now unnecessary `.String()` calls. :)

It does mean you're only guaranteed to get the default game type if you use `NewDisputeGameType()` rather than constructing the enum directly, though currently the default also happens to be the value 0 so it works either way.